docs/csharp/whats-new/csharp-10.md

```diff
@@ -0,0 +1,66 @@
+
+C# 10.0 adds the following features and enhancements to the C# language (as of .NET 6 Preview 5):
+
+- Allow `const` interpolated strings.
+- Record types can seal `ToString()`.
+- Allow both assignment and declaration in the same deconstruction.
+- Allow `AsyncMethodBuilder` attribute on methods.
+
+Some of the features you can try are available only when you set your language version to "preview". These features may have more refinements in future previews before .NET 6.0 is released.
+
+C# 10.0 is supported on **.NET 6**. For more information, see [C# language versioning](../language-reference/configure-language-version.md).
+
+You can download the latest .NET 6.0 SDK from the [.NET downloads page](https://dotnet.microsoft.com/download). You can also download [Visual Studio 2022 preview](https://visualstudio.microsoft.com/vs/preview/vs2022/), which includes the .NET 6.0 preview SDK.
+
+## Constant interpolated strings
+
+In C# 10.0, `const` strings may be initialized using [string interpolation](../language-reference/tokens/interpolated.md) if all the placeholders are themselves constant strings. String interpolation can create more readable constant strings as you build constant strings used in your application. The placeholder expressions can't be numeric constants because those constants are converted to strings at runtime. The current culture may affect their string representation. Learn more in the language reference on [`const` expressions](../language-reference/keywords/const.md).
+
+> [!NOTE]
+> When using .NET 6.0 preview 5, this feature requires setting the `<LangVersion>` element in your *csproj* file to `preview`.
+
+## Record types can seal ToString
+
+In C# 10.0, you can add the `sealed` modifier when you override `ToString` in a record type. Sealing the `ToString` method prevents the compiler from synthesizing a `ToString` method for any derived record types. This ensures all derived record types use the `ToString` method defined in a common base record type. You can learn more about this feature in the article on [records](../language-reference/builtin-types/record.md).
+
+> [!NOTE]
+> When using .NET 6.0 preview 5, this feature requires setting the `<LangVersion>` element in your *csproj* file to `preview`.
+
+## Assignment and declaration in same deconstruction
+
+This change removes a restriction from earlier versions of C#. Previously, a deconstruction could assign all values to existing variables, or initialize newly declared variables:
+
+```csharp
+// Initialization:
+(int x, int y) = point;
+
+// assignment:
+int x1 = 0;
+int y1 = 0;
+(x1, y1) = point;
+```
+
+C# 10.0 removes this restriction:
+
+```csharp
+int x = 0;
+(x, int y) = point;
+```
+
+> [!NOTE]
+> When using .NET 6.0 preview 5, this feature requires setting the `<LangVersion>` element in your *csproj* file to `preview`.
+
+## Allow AsyncMethodBuilder attribute on methods
```

Comment:
This is the only section that doesn't warn that you have to set LangVersion to preview -- is that omission intended?

Reply:
yes, that it intentional.